packages/interfaces/src/sender.ts

```diff
@@ -1,10 +1,23 @@
 import type { IEncoder, IMessage } from "./message.js";
-import { ProtocolUseOptions, SDKProtocolResult } from "./protocols.js";
+import { SDKProtocolResult } from "./protocols.js";
+
+export type ISenderOptions = {
+  /**
+   * Enables retry of a message that was failed to be sent.
+   * @default false
+   */
+  autoRetry?: boolean;
+  /**
+   * Sets number of attempts if `autoRetry` is enabled.
+   * @default 3
+   */
+  maxAttempts?: number;
+};
 
 export interface ISender {
   send: (
     encoder: IEncoder,
     message: IMessage,
-    sendOptions?: ProtocolUseOptions
+    sendOptions?: ISenderOptions
```

Comment:
this might be a breaking change since `ProtocolUseOptions` had a variable not in `ISenderOptions`. If someone has code where that optional variable was set, will it break upon updating?

Reply:
true and no, `ISenderOptions` actually returns previously existed properties and now they are respected in this implementation 

so the only breaking change is the naming

Reply:
new interface doesn't have `forceUseAllPeers?: boolean;`

Reply:
https://github.com/waku-org/js-waku/blob/b2efce5ec27807325685cc32f9333805e6321ac7/packages/interfaces/src/protocols.ts#L41

Reply:
good spot, afaik it wasn't used outside some of our tests